### PR TITLE
allow running of simulations for full Li metal batteries

### PR DESCRIPTION
**Pull request by molel-gt**
# Description

Allow running of simulations for full Li metal batteries.

Fixes # 1659

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by tinosulzer — COMMENTED** [review]
There is nothing about solid state in this PR? Lithium metal is different from solid state

File: pybamm/simulation.py [hunk 1]
```diff
@@ -84,13 +84,6 @@ def __init__(
     ):
         self.parameter_values = parameter_values or model.default_parameter_values
 
-        if isinstance(model, pybamm.lithium_ion.BasicDFNHalfCell):
-            if experiment is not None:
-                raise NotImplementedError(
-                    "BasicDFNHalfCell is not compatible "
-                    "with experiment simulations yet."
-                )
-
```
[the remark this answers is not in the document]
Comment (molel-gt) [reply pybamm/simulation.py]: Thank you. I have added back the exception clauses and tests

**Review by molel-gt — COMMENTED** [review]
> There is nothing about solid state in this PR? Lithium metal is different from solid state

Yes. I have removed the reference.

**Review by tinosulzer — COMMENTED** [review]
Have you tried running this locally? It shouldn't work, I removed `pybamm.lithium_metal_electrode.Full`. Just use the half-cell DFN.

**Comment by molel-gt** [comment]
> Have you tried running this locally? It shouldn't work, I removed `pybamm.lithium_metal_electrode.Full`. Just use the half-cell DFN.

@tinosulzer actually, I am getting it to run locally, but I have my local copy of parameters file.

**Pull request closed without merging**